src/west/commands.py

```diff
@@ -246,15 +246,32 @@ def git_version_info(self):
         '''
         if not hasattr(self, '_git_ver'):
             self.die_if_no_git()
-            # raw_ver usually looks like '2.31.1'
-            raw_ver = self.check_output(
-                [self._git, '--version']).decode().strip().split()[-1]
-            # match just the numeric prefix, in case we get something like
-            # '2.31.1-EXTRAVERSION' in some situations. The major/minor/patch
-            # group names are purely for clarity.
-            match = re.match(
-                r'^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)', raw_ver)
-            self._git_ver = tuple(int(group) for group in match.groups())
+            # The raw 'git --version' output varies by platform and due to
+            # i18n. Examples in English are:
+            #
+            # git version 2.25.1
+            # git version 2.28.0.windows.1
+            # git version 2.24.3 (Apple Git-128)
+            #
+            # We handle this by matching the first string in the output
+            # that has a prefix that looks like a semver. Hopefully this
+            # is general enough.
+
+            raw_ver = self.check_output([self._git, '--version']).decode()
+
+            # Match just the numeric prefix, to handle '.windows.1'
+            # type suffixes.
+            ver_re = re.compile(r'^(\d+)\.(\d+)\.(\d+)')
```

Comment:
Looks like this `^` has no effect when using `match()`, only when using `search()` (ignoring MULTILINE).

Reply:
Should the last digit be optional (`?`) to be on the safe side?

Reply:
> Looks like this `^` has no effect when using `match()`, only when using `search()` (ignoring MULTILINE).
> Should the last digit be optional (?) to be on the safe side?

Good catch on both fronts. I did some more digging in the git sources and pushed a new version that IMO handles all the possibilities we can reasonably expect.